### PR TITLE
Make KubeRay Operator Image FIPS compliant

### LINKED ISSUE #1632: Make KubeRay Operator Image FIPS compliant
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/kuberay/issues) and found no similar feature requirement.

### Description

The KubeRay Operator image is currently not FIPS compliant, which makes it such that it is unusable in some high security environments.

https://github.com/openshift/check-payload is one tool that can be used to check FIPS compliance.
On running with the latest v1.0.0 image from Quay, I'm seeing the following output:
```
---- Failure Report
+-----------------+------------------------------+---------------------------------+
| EXECUTABLE NAME | STATUS                       | IMAGE                           |
+-----------------+------------------------------+---------------------------------+
|                 | openssl library not present  | quay.io/kuberay/operator:v1.0.0 |
| /manager        | go binary is not CGO_ENABLED | quay.io/kuberay/operator:v1.0.0 |
+-----------------+------------------------------+---------------------------------+
F1106 17:08:00.428070 3316466 main.go:259] Error: run failed
```

cc @kevin85421 

### Use case

I would like to use KubeRay in FIPS environments.

### Related issues

Related Ray issue: https://github.com/ray-project/ray/pull/40755

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by anishasthana**
cc @kevin85421  @tedhtchang @jbusche 

Closes #1632

Test image that I built and pushed:  quay.io/anishasthana/kuberay-operator:ubiminimaltest

Output of payload checker:
```
---- Successful run
```

### CODE REVIEW

**Comment by tedhtchang** [comment]
/LGTM. We need the openssl shared lib from the ubi8-mini image and also need to compile the go binary to dynamically link to these libs.

**Comment by jbusche** [comment]
Hi @anishasthana I tried a test of your PR, and it looks good to me with CodeFlare.  I manually built an image using `make docker-image` and then did the following tests:
1.  Tested the check-payload scan and it looked good:
```
./check-payload scan operator --spec quay.io/jbusche/pr-1633:anish
---- Successful run
```
2.  I did a Twistlock scan, and it looks good, except the the go 1.19 vulnerabilities that continue to persist... nothing new added or changed.
```
vulnerabilityFailureSummary","C:1|H:0|M:2|L:0|T:3"
```
[twistlock-report-pr-1633_anish-20231107-00-01.results.csv](https://github.com/ray-project/kuberay/files/13274238/twistlock-report-pr-1633_anish-20231107-00-01.results.csv)

3.  I tried deploying the built image on my OpenShift Cluster and running CodeFlare against it.  The raycluster and underlying codeflare-sdk examples looked good.
```
oc get raycluster -n default
NAME      DESIRED WORKERS   AVAILABLE WORKERS   STATUS   AGE
jobtest   2                 2                   ready    9m51s

and:
AppStatus:
  msg: !!python/object/apply:ray.dashboard.modules.job.common.JobStatus
  - SUCCEEDED
 ```

File: ray-operator/Dockerfile [hunk 1]
```diff
@@ -16,9 +16,9 @@ COPY controllers/ controllers/
 
 # Build
 USER root
-RUN CGO_ENABLED=0 GOOS=linux GO111MODULE=on go build -a -o manager main.go
+RUN CGO_ENABLED=1 GOOS=linux go build -tags strictfipsruntime -a -o manager main.go
```
Comment (kevin85421) [inline ray-operator/Dockerfile:19]: For my education, would you mind explaining the change?

Comment (kevin85421) [reply ray-operator/Dockerfile:19]: Thank you for the explanation!

**Review by kevin85421 — APPROVED** [review]

**Pull request merged**